classes/Topic.php

```diff
@@ -0,0 +1,339 @@
+<?php
+/**
+ * Topic
+ *
+ * @package TheyWorkForYou
+ */
+
+namespace MySociety\TheyWorkForYou;
+
+class Topic {
+
+    /**
+     * DB handle
+     */
+    private $db;
+
+    private $raw;
+    private $id;
+    private $title;
+    private $slug;
+    private $description;
+    private $image;
+    private $search_string;
+    private $front_page;
+    private $actions;
+
+    /**
+     * Constructor
+     *
+     */
+
+    public function __construct($data = NULL)
+    {
+        $this->db = new \ParlDB;
+
+        if (is_null($data)) {
+          return;
+        }
+
+        $this->raw = $data;
+        if (isset($data['id'])) {
+            $this->id = $data['id'];
+        }
+        $this->title = $data['title'];
+        $this->slug = $data['slug'];
+        $this->description = $data['description'];
+        $this->image = $data['image'];
+        if (isset($data['search_string'])) {
```

Comment:
As far as I can see, this constructor is always called with this and `front_page` being present, if you've got here.
And ID too in fact, it's never called with any data unless it's all of it.